### ISSUE TITLE
Support for stable-diffusion-3-medium model

### ISSUE THREAD

**Issue by rupeshs** [huggingface/diffusers#8482]
**Is your feature request related to a problem? Please describe.**
No.

**Describe the solution you'd like.**
Support SD3 medium model (https://huggingface.co/stabilityai/stable-diffusion-3-medium).

**Describe alternatives you've considered.**
It seems like this PR(https://github.com/huggingface/diffusers/issues/7232) is stale

**Comment by sayakpaul** [comment]
Coming up in some hours ;)

**Comment by tin2tin** [comment]
https://huggingface.co/stabilityai/stable-diffusion-3-medium/commit/fc46bd4cf48394d33a813fe81f4241568cc8b85b

![image](https://github.com/huggingface/diffusers/assets/1322593/03c9eae5-5112-49ef-a502-88f976a2aea1)

So, do we need to wait for an update of Diffusers?

**Comment by painebenjamin** [comment]
> https://huggingface.co/stabilityai/stable-diffusion-3-medium/commit/fc46bd4cf48394d33a813fe81f4241568cc8b85b
> 
> ![image](https://private-user-images.githubusercontent.com/1322593/339027435-03c9eae5-5112-49ef-a502-88f976a2aea1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgyMDYzMDYsIm5iZiI6MTcxODIwNjAwNiwicGF0aCI6Ii8xMzIyNTkzLzMzOTAyNzQzNS0wM2M5ZWFlNS01MTEyLTQ5ZWYtYTUwMi04OGY5NzZhMmFlYTEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MTJUMTUyNjQ2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTYxZDBhOTg1MzMyMjA5MTI4YTVjNjk4OTZlN2E2YTJjNDliNDg5NDY0OTUxMTgxY2ZlMDQ2NmYxMzJlZmQ0ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.V01Mp99QV9Y_8pSDyDduDDC1BC65S3fwXAhi9K60AHg)
> 
> So, do we need to wait for an update of Diffusers?

At some point in the near future the SD3 branch will be merged into diffusers which will add support for it. See [this](https://github.com/huggingface/diffusers/pull/8483) pull request for details.

At the time of writing, diffusers-formatted weights (and control files like `model_index.json`) are _not_ available at the huggingface repository, so even if you pull that branch, it will not work yet. Give it a few hours and that will likely change.

**Comment by sayakpaul** [comment]
We will let everyone know once y'all are good to go. Till then, have patience :)

**Comment by tin2tin** [comment]
The files for diffusers are up, but gated files will only be accessible on the Huggingface platform, as there is no way, afaik, to get access when running diffusers locally, and only downloading it with diffusers from HF.  
https://huggingface.co/stabilityai/stable-diffusion-3-medium-diffusers

**Comment by painebenjamin** [comment]
We can use the code from [the HF space](https://huggingface.co/spaces/stabilityai/stable-diffusion-3-medium/blob/main/app.py) to run it without needing to use that repository.

```sh
pip install https://github.com/huggingface/diffusers.git@sd3
```

Then...

```py
import torch
from diffusers import StableDiffusion3Pipeline

pipe = StableDiffusion3Pipeline.from_pretrained(
    "stabilityai/stable-diffusion-3-medium",
    revision="refs/pr/26",
    torch_dtype=torch.float16,
)
pipe.to("cuda")

image = pipe(
    prompt="a photo of a cat holding a sign that says hello world",
    negative_prompt="",
    num_inference_steps=28,
    height=1024,
    width=1024,
    guidance_scale=7.0,
).images[0]

image.save("sd3_hello_world.png")
```

Happy generating!

**Comment by haofanwang** [comment]
You guys are awesome!

**Comment by rolux** [comment]
> Happy generating!

Thanks a lot!

**Comment by OlegRuban-ai** [comment]
![1](https://github.com/huggingface/diffusers/assets/61088564/0cf29b8c-6b4c-403b-91cc-09767a92cf49)
Hello! Is it possible to do something with optimization? On Comfy it runs with 12 GB GPU, and with diffusers - 21.

**Comment by sayakpaul** [comment]
Yeah. The documentation includes all optimization related bits :)

**Comment by tin2tin** [comment]
Hmm... I can't get access to the gated files, even though I accepted everything, and I am logged in, but trying to access them from my local computer through Diffusers just doesn't remove the gate blocking. Neither does accessing a mirrored single file, I can download it, but because of the gate, it doesn't work. Gated files does not play well with Diffusers at all.

**Comment by s9anus98a** [comment]
> We can use the code from [the HF space](https://huggingface.co/spaces/stabilityai/stable-diffusion-3-medium/blob/main/app.py) to run it without needing to use the repository that is gated behind SAI's approval. You will still need to be logged into Huggingface and request access to the first repository, but you shouldn't have to wait for approval.
> 
> ```shell
> pip install git+https://github.com/huggingface/diffusers.git@sd3
> ```
> 
> Then...
> 
> ```python
> import torch
> from diffusers import StableDiffusion3Pipeline
> 
> pipe = StableDiffusion3Pipeline.from_pretrained(
>     "stabilityai/stable-diffusion-3-medium",
>     revision="refs/pr/26",
>     torch_dtype=torch.float16,
> )
> pipe.to("cuda")
> 
> image = pipe(
>     prompt="a photo of a cat holding a sign that says hello world",
>     negative_prompt="",
>     num_inference_steps=28,
>     height=1024,
>     width=1024,
>     guidance_scale=7.0,
> ).images[0]
> 
> image.save("sd3_hello_world.png")
> ```
> 
> Happy generating!

can anyone run sd3 medium on colab free? i tried and it take out all the space. 78GB. WTF??

**Comment by rohitgandikota** [comment]
> > We can use the code from [the HF space](https://huggingface.co/spaces/stabilityai/stable-diffusion-3-medium/blob/main/app.py) to run it without needing to use the repository that is gated behind SAI's approval. You will still need to be logged into Huggingface and request access to the first repository, but you shouldn't have to wait for approval.
> > ```shell
> > pip install git+https://github.com/huggingface/diffusers.git@sd3
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Then...
> > ```python
> > import torch
> > from diffusers import StableDiffusion3Pipeline
> > 
> > pipe = StableDiffusion3Pipeline.from_pretrained(
> >     "stabilityai/stable-diffusion-3-medium",
> >     revision="refs/pr/26",
> >     torch_dtype=torch.float16,
> > )
> > pipe.to("cuda")
> > 
> > image = pipe(
> >     prompt="a photo of a cat holding a sign that says hello world",
> >     negative_prompt="",
> >     num_inference_steps=28,
> >     height=1024,
> >     width=1024,
> >     guidance_scale=7.0,
> > ).images[0]
> > 
> > image.save("sd3_hello_world.png")
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Happy generating!
> 
> can anyone run sd3 medium on colab free? i tried and it take out all the space. 78GB. WTF??

I am trying locally and it takes around 20GB with float16 precision - you can go even low with bfloat16!

**Comment by rolux** [comment]
> Yeah. The documentation includes all optimization related bits :)

Currently, loading the T5 text encoder separately fails with

```
OSError: stabilityai/stable-diffusion-3-medium-diffusers does not appear to have a file named text_encoder_3/model-00004-of-00004.safetensors. Checkout 'https://huggingface.co/stabilityai/stable-diffusion-3-medium-diffusers/tree/main' for available files.
```

... but you're probably aware of that ;)

**Comment by rolux** [comment]
> > Yeah. The documentation includes all optimization related bits :)
> 
> Currently, loading the T5 text encoder separately fails with
> 
> ```
> OSError: stabilityai/stable-diffusion-3-medium-diffusers does not appear to have a file named text_encoder_3/model-00004-of-00004.safetensors. Checkout 'https://huggingface.co/stabilityai/stable-diffusion-3-medium-diffusers/tree/main' for available files.
> ```
> 
> ... but you're probably aware of that ;)

Seems to be fixed in the latest revision.

**Comment by rupeshs** [comment]
SD3 for low VRAM systems: https://github.com/rupeshs/sd3-low-vram
![sd3-lowvram](https://github.com/huggingface/diffusers/assets/3255994/b64d6152-5618-42d1-8a6a-a420c2b54f47)

**Comment by DN6** [comment]
Closing as completed.
